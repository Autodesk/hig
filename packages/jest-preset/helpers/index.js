/* eslint-disable global-require */
module.exports = {
  takeSnapshotsOf: require("./takeSnapshotsOf")
};
