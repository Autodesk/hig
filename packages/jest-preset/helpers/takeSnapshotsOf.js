const React = require("react");
const renderer = require("react-test-renderer");

module.exports = function takeSnapshotsOf(Component, cases, rendererOptions) {
  cases.forEach(({ desc, props }) => {
    it(desc, () => {
      const vDom = React.createElement(Component, props);
      const tree = renderer.create(vDom, rendererOptions).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
};
