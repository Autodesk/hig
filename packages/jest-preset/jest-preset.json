{
  "collectCoverageFrom": ["**/src/**/*.{js,jsx}"],
  "coverageDirectory": "<rootDir>/coverage",
  "coveragePathIgnorePatterns": [
    "__gemini__",
    "__stories__",
    "docs",
    "examples/redux",
    "node_modules",
    "packages/jest-preset",
    "packages/vanilla",
    "src/playground"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 50,
      "functions": 50,
      "lines": 50,
      "statements": 50
    }
  },
  "moduleNameMapper": {
    "\\.(css|scss|svg)$": "@hig/jest-preset/fileMock.js"
  },
  "modulePaths": ["node_modules"],
  "moduleFileExtensions": ["js", "jsx", "json"],
  "setupFiles": ["raf/polyfill"],
  "setupTestFrameworkScriptFile": "@hig/jest-preset/setupTests.js",
  "testPathIgnorePatterns": [
    "docs",
    "packages/babel-preset",
    "packages/jest-preset",
    "packages/vanilla"
  ],
  "unmockedModulePathPatterns": ["node_modules"]
}
