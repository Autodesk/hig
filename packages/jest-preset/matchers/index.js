require("./toBeConstants");
require("./toImplementHIGInterfaceOf");
require("./toHavePropTypesAndDocgenInfo");
