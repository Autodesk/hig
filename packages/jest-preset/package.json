{
  "private": true,
  "name": "@hig/jest-preset",
  "version": "0.1.0",
  "devDependencies": {
    "babel-core": "^6.26.3",
    "babel-jest": "^23.4.0",
    "enzyme": "^2.9.1",
    "jest": "^21.2.1",
    "jest-enzyme": "^3.8.3",
    "jest-extended": "^0.6.0",
    "raf": "^3.4.0",
    "react-test-renderer": "^15.6.1"
  },
  "eslintConfig": {
    "extends": "@hig"
  },
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
