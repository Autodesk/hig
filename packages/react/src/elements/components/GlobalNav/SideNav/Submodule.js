import React, { Component } from "react";
import PropTypes from "prop-types";
import SubmoduleAdapter from "../../../../adapters/GlobalNav/SideNav/SubmoduleAdapter";

class Submodule extends Component {
  static propTypes = {
    id: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired
  };

  handleClick = event => {
    event.preventDefault();
    this.props.onClick(this.props.id);
  };

  render() {
    const { onClick, id, ...otherProps } = this.props;
    return <SubmoduleAdapter onClick={this.handleClick} {...otherProps} />;
  }
}

export default Submodule;
