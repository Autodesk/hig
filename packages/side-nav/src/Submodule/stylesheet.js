export default function stylesheet(props, themeData) {
  return {
    wrapper: {
      fontFamily: themeData[`typography.body.fontFamily`],
      fontSize: themeData[`typography.body.fontSize`],
      fontWeight: themeData[`typography.body.fontWeight`],
      lineHeight: themeData[`typography.body.lineHeight`],
      display: "flex",
      alignItems: "center",
      justifyContent: "space-between",
      paddingBottom: "10px",
      paddingRight: themeData["density.spacings.medium"],
      textDecoration: "none",
      cursor: "pointer",
      userSelect: "none",
      color: themeData["typography.body.color"],

      "&:first-of-type": {
        marginTop: themeData["density.spacings.extraSmall"]
      },

      "&:last-of-type": {
        marginBottom: themeData["density.spacings.medium"]
      },

      "&:focus, &:hover": {
        outline: "none",
        color: themeData["colorScheme.reference.accent"]
      },

      externalIcon: {
        display: "flex",
        alignItems: "center",
        marginLeft: "auto",
        padding: "6px 10px",

        "& svg": {
          fill: themeData["typography.body.color"]
        },

        "&:hover svg *": {
          fill: themeData["colorScheme.reference.accent"]
        }
      }
    }
  };
}
