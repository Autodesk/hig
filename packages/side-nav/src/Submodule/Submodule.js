import React, { Component } from "react";
import PropTypes from "prop-types";
import { css, cx } from "emotion";
import { sizes as iconSizes } from "@hig/icons";
import ThemeContext from "@hig/theme-context";
import {
  createCustomClassNames,
  memoizeCreateButtonEventHandlers
} from "@hig/utils";

import { AVAILABLE_TARGETS, targets } from "../targets";
import ExternalLinkIcon from "../presenters/ExternalLinkIcon";
import stylesheet from "./stylesheet";

export default class Submodule extends Component {
  static propTypes = {
    /** Indicates this submodule is currently active */
    /* eslint-disable react/no-unused-prop-types */
    active: PropTypes.bool,
    /** URL to navigate to when clicking this submodule */
    link: PropTypes.string,
    /** Called when clicking on the submodule */
    onClick: PropTypes.func,
    /** Called when link is focused  */
    onFocus: PropTypes.func,
    /** Called when hovering over the submodule */
    onMouseOver: PropTypes.func,
    /** Function to modify the component's styles */
    stylesheet: PropTypes.func,
    /** Anchor target. Applicable only if link is provided */
    target: PropTypes.oneOf(AVAILABLE_TARGETS),
    /** Text to render */
    title: PropTypes.string
  };

  createEventHandlers = memoizeCreateButtonEventHandlers();

  render() {
    const {
      link,
      onClick,
      onFocus,
      onMouseOver,
      stylesheet: customStylesheet,
      target,
      title,
      ...otherProps
    } = this.props;
    const { className } = otherProps;
    const { handleClick, handleKeyDown } = this.createEventHandlers(onClick, {
      // Allow default on hyperlinks to trigger navigation
      preventDefault: !link
    });
    const Wrapper = link ? "a" : "div";
    const isExternalLink = link && target === targets.BLANK;
    const role = link ? undefined : "button";
    const wrapperTarget = link ? target : undefined;
    const externalIconClassName = createCustomClassNames(
      className,
      "external_icon"
    );

    return (
      <ThemeContext.Consumer>
        {({ resolvedRoles, metadata }) => (
          <Wrapper
            className={cx([
              css(
                stylesheet(
                  { stylesheet: customStylesheet, ...this.props },
                  resolvedRoles
                ).wrapper
              ),
              className
            ])}
            href={link}
            onClick={handleClick}
            onFocus={onFocus}
            onKeyDown={handleKeyDown}
            onMouseOver={onMouseOver}
            role={role}
            tabIndex="0"
            target={wrapperTarget}
          >
            {title}
            {isExternalLink ? (
              <div
                className={cx([
                  css(stylesheet(this.props, resolvedRoles).externalIcon),
                  externalIconClassName
                ])}
              >
                <ExternalLinkIcon
                  size={
                    metadata.densityId === "medium-density"
                      ? iconSizes.PX_24
                      : iconSizes.PX_16
                  }
                />
              </div>
            ) : null}
          </Wrapper>
        )}
      </ThemeContext.Consumer>
    );
  }
}
