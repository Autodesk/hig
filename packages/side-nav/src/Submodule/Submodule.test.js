import { takeSnapshotsOf } from "@hig/jest-preset/helpers";

import Submodule from "./Submodule";

describe("side-nav/Submodule", () => {
  describe("snapshot tests", () => {
    takeSnapshotsOf(Submodule, [
      {
        description: "renders with no props",
        props: {}
      },
      {
        description: "renders with minimal props",
        props: {
          title: "Module"
        }
      },
      {
        description: "renders an external link",
        props: {
          link: "http://example.com",
          target: "_blank",
          title: "Module"
        }
      },
      {
        description: "renders a link",
        props: {
          link: "http://example.com",
          title: "Module"
        }
      },
      {
        description: "renders with a click handler",
        props: {
          onClick: function handleClick() {}
        }
      }
    ]);
  });
});
