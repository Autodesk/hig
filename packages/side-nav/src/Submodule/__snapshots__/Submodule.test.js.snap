// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`side-nav/Submodule snapshot tests renders a link 1`] = `
.emotion-0 {
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 10px;
  padding-right: 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #3c3c3c;
}

.emotion-0:first-of-type {
  margin-top: 8px;
}

.emotion-0:last-of-type {
  margin-bottom: 16px;
}

.emotion-0:focus,
.emotion-0:hover {
  outline: none;
  color: #0696d7;
}

.emotion-0 externalIcon {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: auto;
  padding: 6px 10px;
}

.emotion-0 externalIcon svg {
  fill: #3c3c3c;
}

.emotion-0 externalIcon:hover svg * {
  fill: #0696d7;
}

<a
  className="emotion-0"
  href="http://example.com"
  onClick={undefined}
  onFocus={undefined}
  onKeyDown={undefined}
  onMouseOver={undefined}
  role={undefined}
  tabIndex="0"
  target={undefined}
>
  Module
</a>
`;

exports[`side-nav/Submodule snapshot tests renders an external link 1`] = `
.emotion-2 {
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 10px;
  padding-right: 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #3c3c3c;
}

.emotion-2:first-of-type {
  margin-top: 8px;
}

.emotion-2:last-of-type {
  margin-bottom: 16px;
}

.emotion-2:focus,
.emotion-2:hover {
  outline: none;
  color: #0696d7;
}

.emotion-2 externalIcon {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: auto;
  padding: 6px 10px;
}

.emotion-2 externalIcon svg {
  fill: #3c3c3c;
}

.emotion-2 externalIcon:hover svg * {
  fill: #0696d7;
}

.emotion-0 {
  fill: #808080;
  pointer-events: none;
}

.emotion-0 > * {
  fill: #808080;
  pointer-events: none;
}

<a
  className="emotion-2"
  href="http://example.com"
  onClick={undefined}
  onFocus={undefined}
  onKeyDown={undefined}
  onMouseOver={undefined}
  role={undefined}
  tabIndex="0"
  target="_blank"
>
  Module
  <div
    className="emotion-1"
  >
    <svg
      className="emotion-0"
      color={undefined}
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        external-link-24
      </title>
      <path
        d="M18 18H6V6h4V5H5v14h14v-5h-1v4z"
        fillRule="evenodd"
      />
      <path
        d="M11.88 5.05L14 7.17l-3.54 3.54 2.83 2.83L16.83 10 19 12.12l1.41-8.48zm6.36 5l-1.43-1.48-3.53 3.54-1.4-1.4 3.53-3.54L14 5.76l5.15-.91z"
        fillRule="evenodd"
      />
    </svg>
  </div>
</a>
`;

exports[`side-nav/Submodule snapshot tests renders with a click handler 1`] = `
.emotion-0 {
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 10px;
  padding-right: 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #3c3c3c;
}

.emotion-0:first-of-type {
  margin-top: 8px;
}

.emotion-0:last-of-type {
  margin-bottom: 16px;
}

.emotion-0:focus,
.emotion-0:hover {
  outline: none;
  color: #0696d7;
}

.emotion-0 externalIcon {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: auto;
  padding: 6px 10px;
}

.emotion-0 externalIcon svg {
  fill: #3c3c3c;
}

.emotion-0 externalIcon:hover svg * {
  fill: #0696d7;
}

<div
  className="emotion-0"
  href={undefined}
  onClick={[Function]}
  onFocus={undefined}
  onKeyDown={[Function]}
  onMouseOver={undefined}
  role="button"
  tabIndex="0"
  target={undefined}
/>
`;

exports[`side-nav/Submodule snapshot tests renders with minimal props 1`] = `
.emotion-0 {
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 10px;
  padding-right: 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #3c3c3c;
}

.emotion-0:first-of-type {
  margin-top: 8px;
}

.emotion-0:last-of-type {
  margin-bottom: 16px;
}

.emotion-0:focus,
.emotion-0:hover {
  outline: none;
  color: #0696d7;
}

.emotion-0 externalIcon {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: auto;
  padding: 6px 10px;
}

.emotion-0 externalIcon svg {
  fill: #3c3c3c;
}

.emotion-0 externalIcon:hover svg * {
  fill: #0696d7;
}

<div
  className="emotion-0"
  href={undefined}
  onClick={undefined}
  onFocus={undefined}
  onKeyDown={undefined}
  onMouseOver={undefined}
  role="button"
  tabIndex="0"
  target={undefined}
>
  Module
</div>
`;

exports[`side-nav/Submodule snapshot tests renders with no props 1`] = `
.emotion-0 {
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 10px;
  padding-right: 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #3c3c3c;
}

.emotion-0:first-of-type {
  margin-top: 8px;
}

.emotion-0:last-of-type {
  margin-bottom: 16px;
}

.emotion-0:focus,
.emotion-0:hover {
  outline: none;
  color: #0696d7;
}

.emotion-0 externalIcon {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: auto;
  padding: 6px 10px;
}

.emotion-0 externalIcon svg {
  fill: #3c3c3c;
}

.emotion-0 externalIcon:hover svg * {
  fill: #0696d7;
}

<div
  className="emotion-0"
  href={undefined}
  onClick={undefined}
  onFocus={undefined}
  onKeyDown={undefined}
  onMouseOver={undefined}
  role="button"
  tabIndex="0"
  target={undefined}
/>
`;
